arch/platform/cooja/sys/cooja_mt.c

```diff
@@ -36,36 +36,95 @@
  * Fredrik Osterlind <fros@sics.se>
  */
 
-#include "contiki.h"
+#ifdef __APPLE__
+#define _DARWIN_C_SOURCE /* For pthread_get_stackaddr_np() */
+#else
+#define _GNU_SOURCE /* For pthread_getattr_np(). */
+#endif
+#include <pthread.h>
+
 #include "sys/cooja_mt.h"
-#include "sys/cc.h"
+
+#include <signal.h>
+#include <stdio.h>
 
 #define MT_STATE_READY   1
 #define MT_STATE_RUNNING 2
 #define MT_STATE_EXITED  5
 
 static struct cooja_mt_thread *current;
+static struct cooja_mt_thread *cooja_running_thread;
 
+#ifdef __APPLE__
+/* Avoid deprecated warnings for ucontext.h functions on OS X. */
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
+#endif
+
+/*--------------------------------------------------------------------------*/
+int
+cooja_mt_init(struct cooja_mt_thread *t)
+{
+#ifdef __APPLE__
+  /* Apple's makecontext() fails if size is less than MINSIGSTKSZ. */
+  if(COOJA_MTARCH_STACKSIZE < MINSIGSTKSZ) {
+    return -1;
+  }
+#endif
+  void *stackAddr;
+  size_t stackSize;
+  pthread_t self = pthread_self();
+#ifdef __APPLE__
+  stackAddr = pthread_get_stackaddr_np(self);
+  stackSize = pthread_get_stacksize_np(self);
+#else
+  pthread_attr_t attrs;
+  if(pthread_getattr_np(self, &attrs) != 0) {
+    return -2;
+  }
+  if(pthread_attr_getstack(&attrs, &stackAddr, &stackSize) != 0) {
+    return -3;
+  }
+#endif
+  getcontext(&t->ctxt);
```

Comment:
Please check the return value from `getcontext()` for errors, as is done in other places.

Reply:
Fixed.